src/components/UpgradePurchase/UpgradePurchase.js

```diff
@@ -0,0 +1,95 @@
+import React, { Component } from 'react'
+import classNames from 'classnames'
+import Button from '@material-ui/core/Button'
+import Card from '@material-ui/core/Card'
+import CardHeader from '@material-ui/core/CardHeader'
+import CardActions from '@material-ui/core/CardActions'
+import { array, func, number, object } from 'prop-types'
+
+import { IngredientsList } from '../Recipe/Recipe'
+
+import { canMakeRecipe, doesInventorySpaceRemain } from '../../utils'
+
+import { craftedItems } from '../../img'
+
+import FarmhandContext from '../../Farmhand.context'
+import './UpgradePurchase.sass'
+
+export class UpgradePurchase extends Component {
```

Comment:
My preference would be use function-based components where possible (to stay consistent with current React best practices), but I'm curious if there was a specific reason you chose to go with a class-based component here?

Reply:
no _good_ reason 😄 . i started with a copy of `TierPurchase`, and i just tend to enjoy class components. will update!